app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -131,6 +170,107 @@ private void showFilterMenu(View view) {
         });
 
         popupMenu.show();
+
+    }
+
+
+    // TODO: This is an interface method for a dialog fragment. Useless for now since I can't figure out how to have a dialog pop up inside a fragment
```

Comment:
```AddEditExpenseFragment.newInstance(expenseDataList.get(pos)).show(getSupportFragmentManager(), "EDIT_EXPENSE"))``` This is how I made a dialog fragment popup for expbook. Not sure if it helps

Reply:
I see. I assume the code you have here is inside your MainActivity file? Because I did do something similar in expbook too.
The issue here is that I implemented the dialog listener in HomeFragment and I get a runtime exception that MainActivity doesn't implement the listener interface. So I have to figure out a way to make it so that HomeFragment is the context for my dialog listener. I think jared did something like this where he made a callback. Will look into it.

Reply:
Ahhh, yep. That was in the mainactivity. Yep Callback interface is porbably a good idea